### PR TITLE
Use and require Java 22

### LINKED ISSUE #440: Use and require Java 22
Stay in sync with Trino Java version.
Currently, we can't use the latest Trino as it requires Java 22 from version 447 onward.

### PR DESCRIPTION
**Pull request by oneonestar**
## Description
Close #440 
**Do not merge yet.**

## Release notes
(x) Release notes are required, with the following suggested text:

```markdown
* Require Java 22 for build and runtime.
```

### CODE REVIEW

**Review by mosabua — CHANGES_REQUESTED** [review]
Good start. I think we also need to update the docker container build and quickstart docs though. Also maybe jvm config?

**Comment by oneonestar** [comment]
Docker container build refers to `.java-version` file, which has been updated.
I didn't find settings related to Java 21/22 in the quickstart or docs.

**Review by mosabua — APPROVED** [review]
Looks good to me. Lets get other maintainers to confirm as well

File: pom.xml [hunk 1]
```diff
@@ -39,11 +39,13 @@
...
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-        <project.build.targetJdk>21</project.build.targetJdk>
-        <air.java.version>21</air.java.version>
+        <project.build.targetJdk>22</project.build.targetJdk>
+        <air.java.version>22</air.java.version>
         <air.check.skip-spotbugs>true</air.check.skip-spotbugs>
         <air.check.skip-pmd>true</air.check.skip-pmd>
         <air.release.preparation-goals>clean verify -DskipTests</air.release.preparation-goals>
+        <dep.trino.version>454</dep.trino.version>
+        <dep.errorprone.version>2.29.2</dep.errorprone.version>
```
Comment (ebyhr) [inline pom.xml:48]: Please order alphabetically.

Comment (Chaho12) [reply pom.xml:48]: at first i thought it was wierd that ide does not check this
but i then notice that this `properties` section itself is not ordered

wierd. i dont think sortpom checkstyle doesn't check this at all 
but it saids to sort plugin versions https://github.com/airlift/airbase/blob/master/airbase/pom.xml#L164

**Review by Chaho12 — APPROVED** [review]
LGTM

**Comment by wendigo** [comment]
Nice!

**Comment by mosabua** [comment]
@willmostly @vishalya ?

**Review by wendigo — APPROVED** [review]

**Review by willmostly — APPROVED** [review]
LGTM!

**Comment by mosabua** [comment]
Got confirmation from @vishalya and @andythsu on slack as well. Merging.

**Pull request merged**